### PR TITLE
PE-6941 add description and keywords

### PR DESCRIPTION
**Pull request by vilenarios**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,6 +201,24 @@ Sets the ticker symbol for the ANT.
...
 | Ticker   | string | N/A          | true     | New ticker symbol for ANT.        |
 
+#### `Set-Description`
+
+Sets the description for the ANT.
+
+| Tag Name    | Type   | Pattern           | Required | Description                       |
+| ----------- | ------ | ----------------- | -------- | --------------------------------- |
+| Action      | string | "Set-Description" | true     | Action tag for triggering handler |
+| Description | string | N/A               | true     | New description for ANT.          |
```
[the remark this answers is not in the document]
Comment (vilenarios) [reply README.md:211]: Added 512 character pattern.

File: README.md [hunk 2]
```diff
@@ -201,6 +201,24 @@ Sets the ticker symbol for the ANT.
...
+
+#### `Set-Keywords`
+
+Sets the keywords for the ANT.
+
+| Tag Name | Type   | Pattern        | Required | Description                       |
+| -------- | ------ | -------------- | -------- | --------------------------------- |
+| Action   | string | "Set-Keywords" | true     | Action tag for triggering handler |
+| Keywords | table  | "^[%w-_]+$"    | true     | New keywords for ANT.             |
```
[the remark this answers is not in the document]
Comment (vilenarios) [reply README.md:220]: Added # and @.

File: README.md [hunk 3]
```diff
@@ -201,6 +201,24 @@ Sets the ticker symbol for the ANT.
...
+Sets the description for the ANT.
+
+| Tag Name    | Type   | Pattern           | Required | Description                       |
+| ----------- | ------ | ----------------- | -------- | --------------------------------- |
+| Action      | string | "Set-Description" | true     | Action tag for triggering handler |
+| Description | string | N/A               | true     | New description for ANT.          |
+
+#### `Set-Keywords`
+
+Sets the keywords for the ANT.
```
[the remark this answers is not in the document]
Comment (vilenarios) [reply README.md:215]: Updated this to include that detail.

File: src/common/balances.lua [hunk 4]
```diff
@@ -37,4 +37,32 @@ function balances.setTicker(ticker)
...
+function balances.setDescription(description)
+	assert(type(description) == "string", "Description must be a string")
+	assert(#description <= 512, "Description must not be longer than 512 characters")
+	Description = description
+	return json.encode({ description = Description })
+end
+
+function balances.setKeywords(keywords)
+	assert(type(keywords) == "table", "Keywords must be an array")
+	assert(#keywords <= 16, "There must not be more than 16 keywords")
```
[the remark this answers is not in the document]
Comment (vilenarios) [reply src/common/balances.lua:49]: What if they want to empty their keywords?

File: src/common/initialize.lua [hunk 5]
```diff
@@ -26,8 +30,23 @@ function initialize.initializeANTState(state)
...
+	local seenKeywords = {}
+	for _, keyword in ipairs(keywords) do
+		assert(type(keyword) == "string", "Each keyword must be a string")
+		assert(#keyword <= 32, "Each keyword must not be longer than 32 characters")
+		assert(not keyword:find("%s"), "Keywords must not contain spaces")
+		assert(keyword:match("^[%w-_]+$"), "Keywords must only contain alphanumeric characters, dashes, or underscores")
+		assert(not seenKeywords[keyword], "Duplicate keyword detected: " .. keyword)
+		seenKeywords[keyword] = true
+	end
+
```
[the remark this answers is not in the document]
Comment (vilenarios) [reply src/common/initialize.lua:45]: Great point.  Added utility function to validate keywords.

**Review by atticusofsparta — APPROVED** [review]

**Pull request merged**